sogs/routes/dm.py

```diff
@@ -112,3 +110,25 @@ def send_inbox(sid):
     with db.transaction():
         msg = Message(data=utils.decode_base64(message), recip=recip_user, sender=g.user)
     return jsonify(_serialize_message(msg, include_message=False)), http.CREATED
+
+
+@dm.delete("/inbox")
+@auth.blind_user_required
+def delete_inbox_items():
+    """
+    Deletes the user's sent and recieved messages.
+    # Return value
+
+    204 No Content - On successful removal of messages.
+    304 Not Modified - If there were no messages to remove.
```

Comment:
I don't like either of these statuses:

- On success everywhere else in SOGS we return a 200 OK with a json result (even if empty).

- 304 is a redirection response code for cache-control headers which which seems deeply inappropriate for a DELETE request response.  A 200 response code seems appropriate in this case: we asked to delete all, and it succeeded; the fact that there were none to delete doesn't mean it didn't have the intended effect.